### DISCUSSION TITLE
How to have users connect a wallet from a separate provider when signed in with ConnectKit?

### DISCUSSION THREAD

**Discussion by montycheese** [family/connectkit#158] [General]
I'm working on a web application that uses ConnectKit and SIWE to perform login authentication.

We create an off-chain account for the user that stores their authenticated wallet address as well as some other user input fields. One of our use cases is allowing users to linking other wallet addresses to their off-chain user account. 

With connectkit however, there does not seem to be a way of switch the currently connected address to a wallet from a separate wallet provider. (switching between addresses within the active wallet provider works fine however).
Example: User signs in with Metamask Address A. User is signed in. User is able to connect Coinbase Address B to web app and link it to their app account.

What would be the best way to implement this, if it's even possible with ConnectKit?

**Comment by boodtama** [comment]
Would love to know this as well.

**Comment by lochie** [comment]
this would be a question better suited for [wagmi](https://github.com/wagmi-dev/wagmi/discussions), but from what I'm aware wagmi only allows one address connected at a time. but it sounds like you would want to [disconnect](https://wagmi.sh/react/hooks/useDisconnect) the wallet once you've validated they've signed in using [onSignIn](https://docs.family.co/connectkit/usesiwe-hook#options-onsignin), then get them to connect and sign in again with their next wallet and do the same flow.

i would suggest opening a discussion on wagmi to see if they have found any better solutions for off-chain account syncing as someone may have a better solution than the above.
